File: CSharp/Items/MSCE.WPF/UserControl.cs
/*--------------------------------------------------------------------------------------+
|   $safeitemname$.cs
|
+--------------------------------------------------------------------------------------*/

#region System Namespaces
using System;
using System.Windows.Controls;
#endregion

#region Bentley Namespaces
using BDPN = Bentley.DgnPlatformNET;
using BMW = Bentley.MstnPlatformNET.WPF;
using BW = Bentley.Windowing;
#endregion

namespace $rootnamespace$
{
public partial class $safeitemrootname$ : UserControl
{
#region Bentley DockableWindow
private static BMW.DockableWindow $safeitemrootname$DockableWindow { get; set; }

internal void ShowWindow(string unparsed = "")
{
    if (null != $safeitemrootname$DockableWindow)
    {
        $safeitemrootname$DockableWindow.Focus();
        return;
    }

    $safeitemrootname$DockableWindow = new BMW.DockableWindow();
    $safeitemrootname$DockableWindow.Content = new $safeitemrootname$();
    $safeitemrootname$DockableWindow.Attach(Program.Addin, "control", new System.Drawing.Size(Convert.ToInt32($safeitemrootname$DockableWindow.MinWidth),
                Convert.ToInt32($safeitemrootname$DockableWindow.MinHeight)));
    $safeitemrootname$DockableWindow.WindowContent.CanDockVertically = false;
    $safeitemrootname$DockableWindow.WindowContent.ContentCloseQuery += new BW.ContentCloseEventHandler(OnClose);
}

/// <summary>
/// Close and dispose the usercontrol.
/// </summary>
/// <param name="sender"></param>
/// <param name="e"></param>
private static void OnClose(object sender, BW.ContentCloseEventArgs e)
{
    e.CloseAction = BW.ContentCloseAction.Dispose;
    if (null != $safeitemrootname$DockableWindow)
    {
        $safeitemrootname$DockableWindow.Detach();
        $safeitemrootname$DockableWindow.Dispose();
        $safeitemrootname$DockableWindow = null;
    }
}
#endregion
}
}

File: Visual Basic/Items/MSCE.WPF/UserControl.vb
'--------------------------------------------------------------------------------------+
'|   $safeitemname$.vb
'|
'+--------------------------------------------------------------------------------------

#Region "System Namespaces"
Imports System.Windows.Controls
#End Region

#Region "Bentley Namespaces"
Imports BDPN = Bentley.DgnPlatformNET
Imports BG = Bentley.GeometryNET
Imports BM = Bentley.MstnPlatformNET
Imports BMW = Bentley.MstnPlatformNET.WPF
Imports BW = Bentley.Windowing
#End Region

Partial Public Class $safeitemrootname$
        Inherits UserControl

#Region "Bentley DockableWindow"
    Private Shared Property dockableWindow As BMW.DockableWindow

    ''' <summary>
    ''' Show the form if it is not already displayed
    ''' </summary>
    Friend Sub ShowWindow(ByVal Optional unparsed As String = Nothing)
        If dockableWindow IsNot Nothing Then
            dockableWindow.Focus()
            Return
        End If

        dockableWindow = New Bentley.MstnPlatformNET.WPF.DockableWindow()
        dockableWindow.Content = New $safeitemrootname$()
        dockableWindow.Attach(Program.Addin, "control", New Drawing.Size(Convert.ToInt32(dockableWindow.MinWidth), Convert.ToInt32(dockableWindow.MinHeight)))
        dockableWindow.WindowContent.CanDockVertically = False
        AddHandler dockableWindow.WindowContent.ContentCloseQuery, AddressOf OnClose
    End Sub

    ''' <summary>
    ''' Close and dispose the usercontrol.
    ''' </summary>
    ''' <param name="sender"></param>
    ''' <param name="e"></param>
    Private Shared Sub OnClose(sender As Object, e As BW.ContentCloseEventArgs)
        e.CloseAction = BW.ContentCloseAction.Dispose
        dockableWindow.WindowContent.Hide()
        If dockableWindow IsNot Nothing Then
            dockableWindow.Detach()
            dockableWindow.Dispose()
            dockableWindow = Nothing
        End If
    End Sub
#End Region
End Class